### PR TITLE
feat(components/molecule/dropdownList): simplify the props provided

### LINKED ISSUE #2261: 🪲 The autosuggest cannot get selected using arrows and enter button
### URL

https://sui-components.vercel.app/workbench/molecule/autosuggest/demo

### Steps to reproduce

_No response_

### Expected Behavior

_No response_

### Screenshots

_No response_

### Which device have you used?

- [ ] Desktop
- [ ] Tablet
- [ ] Mobile
- [ ] smartTV

### Which operating systems have you used?

- [ ] Windows
- [ ] Linux
- [ ] macOS

### browser and version

_No response_

### Additional info

_No response_

### LINKED ISSUE #2262: 🪲 The handlers must return extended arguments
### URL

https://sui-components.vercel.app/workbench/molecule/autosuggest/demo

### Steps to reproduce

_No response_

### Expected Behavior

_No response_

### Screenshots

_No response_

### Which device have you used?

- [ ] Desktop
- [ ] Tablet
- [ ] Mobile
- [ ] smartTV

### Which operating systems have you used?

- [ ] Windows
- [ ] Linux
- [ ] macOS

### browser and version

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by andresin87**
## Molecule/DropdownList

`❓ Ask`

**TASK**: 

### Description, Motivation and Context

Autocomplete handler selection when enter key pressed on an arrowed option selection.

### Types of changes

- [x] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### CODE REVIEW

File: components/molecule/dropdownList/src/index.js [hunk 1]
```diff
@@ -107,10 +107,8 @@ const MoleculeDropdownList = forwardRef(
           .map((child, index) => (
             <ExtendedChildren
               key={index}
-              index={index}
               value={value}
               onSelect={onSelect}
-              onKeyDown={onKeyDown}
```
Comment (andresin87) [inline components/molecule/dropdownList/src/index.js:113]: This causes the bug mentioned

Comment (jordevo) [reply components/molecule/dropdownList/src/index.js:113]: does that mean (removing passing down that `prop`) that the user can no longer navigate through the options with the keyboard?

**Review by jordevo — APPROVED** [review]
🙌

File: components/molecule/dropdownList/src/index.js (hunk 1 above)
Comment (andresin87) [reply components/molecule/dropdownList/src/index.js:113]: no, he still can. Is fired by its wrapper via event bubbling

This prop was initialy ignored by the component until the prop-spreading iteration, which causes the bug

Comment (jordevo) [reply components/molecule/dropdownList/src/index.js:113]: gotcha! thanks

**Pull request merged**